test/labeling/model/test_label_model.py

```diff
@@ -79,6 +115,115 @@ def test_augmented_L_construction(self):
+        self.assertGreaterEqual(probs.min(), 0.0)
+
+    def test_get_accuracy(self):
+        L = np.array([[1, 2], [1, 0]])
+        label_model = self._set_up_model(L)
+        probs = np.array(
+            [
+                [0.99, 0.01],
+                [0.5, 0.5],
+                [0.9, 0.9],
+                [0.99, 0.01],
+                [0.9, 0.9],
+                [0.5, 0.75],
+                [0.9, 0.9],
+                [0.9, 0.1],
+            ]
+        )
+
+        label_model.m = 2
+        label_model.k = 2
+        label_model.P = torch.Tensor([[0.5, 0.0], [0.0, 0.5]])
+        accs = label_model.get_accuracies(probs=probs)
+        np.testing.assert_array_almost_equal(accs, np.array([0.7, 0.825]))
+
+        label_model.mu = nn.Parameter(label_model.mu_init.clone())
+        accs = label_model.get_accuracies(probs=None)
+        np.testing.assert_array_almost_equal(accs, np.array([0.5, 0.355]))
+
+    def test_build_mask(self):
+
+        L = np.array([[1, 2, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L)
+
+        true_mask = np.array(
+            [
+                [0, 0, 1, 1, 1, 1],
+                [0, 0, 1, 1, 1, 1],
+                [1, 1, 0, 0, 1, 1],
+                [1, 1, 0, 0, 1, 1],
+                [1, 1, 1, 1, 0, 0],
+                [1, 1, 1, 1, 0, 0],
+            ]
+        )
+
+        mask = label_model.mask.numpy()
+        np.testing.assert_array_equal(mask, true_mask)
+
+    def test_init_params(self):
+        L = np.array([[1, 2, 1], [1, 0, 1]])
+        label_model = self._set_up_model(L, class_balance=[0.6, 0.4])
+
+        mu_init = label_model.mu_init.numpy()
+        true_mu_init = np.array(
+            [[1.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.8750], [1.0, 0.0], [0.0, 0.0]]
+        )
+        np.testing.assert_array_equal(mu_init, true_mu_init)
+
+        label_model._set_class_balance(class_balance=[0.3, 0.7], Y_dev=None)
+        label_model._init_params()
+
+        mu_init = label_model.mu_init.numpy()
+        true_mu_init = np.array(
+            [[1.0, 0.0], [0.0, 0.0], [0.0, 0.0], [0.0, 0.5], [1.0, 0.0], [0.0, 0.0]]
+        )
+        np.testing.assert_array_equal(mu_init, true_mu_init)
+
+    def test_predict_proba(self):
+        L = np.array([[1, 2, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L)
+
+        label_model.mu = nn.Parameter(label_model.mu_init.clone())
+        probs = label_model.predict_proba(L)
+
+        true_probs = np.array([[0.99, 0.01], [0.99, 0.01]])
+        np.testing.assert_array_almost_equal(probs, true_probs)
+
+    def test_loss(self):
+        L = np.array([[1, 0, 1], [1, 2, 0]])
+        label_model = self._set_up_model(L)
+        label_model._get_augmented_label_matrix(L, higher_order=True)
+
+        label_model.mu = nn.Parameter(label_model.mu_init.clone() + 0.05)
+        self.assertAlmostEqual(
+            label_model.loss_l2(l2=1.0).detach().numpy().ravel()[0], 0.03
```

Comment:
How do we know these are the right values?

Reply:
Comment about that would be good

Reply:
Wolfram alpha! will add math comments

Reply:
Definitely add comments- esp in tests as people often read to understand what's going on!